src/qibolab/result.py

```diff
@@ -17,50 +15,54 @@ class IntegratedResults:
     Associated with AcquisitionType.INTEGRATION and AveragingMode.SINGLESHOT
     """
 
-    def __init__(self, i: np.ndarray, q: np.ndarray):
-        self.voltage: npt.NDArray[iq] = np.recarray(i.shape, dtype=iq)
-        self.voltage["i"] = i
-        self.voltage["q"] = q
+    def __init__(self, data: np.ndarray):
+        self.voltage: npt.NDArray[np.complex128] = data
+
+    @cached_property
+    def voltage_i(self):
+        """Signal magnitude in volts."""
+        return self.voltage.real
+
+    @cached_property
+    def voltage_q(self):
+        """Signal magnitude in volts."""
+        return self.voltage.imag
```

Comment:
Why do you need to cache this? You're not computing anything, so in the best case it is just consuming a bit more space, in the worst one you'd be fighting with cache invalidation.